debian/rules

```diff
@@ -93,6 +93,12 @@ override_dh_auto_install:
 	install -m0644 packaging/systemd/$(PACKAGE)-management.service debian/$(PACKAGE)-management/lib/systemd/system/$(PACKAGE)-management.service
 	install -m0644 packaging/systemd/$(PACKAGE)-management.default $(DESTDIR)/$(SYSCONFDIR)/default/$(PACKAGE)-management
 
+	# cloudstack-ui
+	mkdir -p $(DESTDIR)/usr/share/$(PACKAGE)-ui
+	cd ui && npm install && npm run build && cd ..
```

Comment:
This assumes that npn is available on the system. However, that's not a guarantee.

The .deb (and .rpm) should have a build depends on 'npm'

Reply:
Yes that adds a dependency much like jdk or mvn. I've added nodejs-12 (lts) in the container images that are used to build packages by blueorangutan. https://hub.docker.com/u/bhaisaab

Reply:
The problem is tha nodejs lts may not be available in default/main repo of the distros. Should we document it on docs and install.md?

Reply:
@rhtyd this looks like a good point to be documented indeed

Reply:
> The problem is that nodejs may not be available in default/main repo of the distros. Should we document it on docs and install.md?

No, we should require it in the .spec and control file as a hard requirement. The build will fail without those, so it shouldn't even start to build.

The control and .spec file should exactly specify what is needed to build the packages. Without those dependencies the build should not initiate

Reply:
Added, but that won't be satisfied automatically chances are somebody will install older version of nodejs.